action.yaml

```diff
@@ -43,14 +39,13 @@ inputs:
     description: 'Run `buf push`'
     required: false
     default: 'false'
+  commit-message:
+    description: 'commit message to use when pushing generated code'
+    required: false
+    default: '[skip ci] buf generated code from protos'
 runs:
   using: "composite"
   steps:
-    - name: Checkout
```

Comment:
Or are we explicitly saying "if you use this new version, checkouts are your responsibility"?

Reply:
It's more flexible, but introduces the requirement that the user uses the checkout action in their workflow before using this action, hence the breaking change. The problem I ran into was running an `npm install` before this action to install some packages used in my buf.gen.yaml, then the second checkout here overwrites node_modules because it checks out the code again and my node_modules are now gone, so my `buf generate` command fails. If we modify our action to not checkout then it allows users to do more in their workflows without getting interfered with.

I suppose we could make it an input, defaulted to true, which would make it a non-breaking change but allow opt-out if you want to check out yourself and do stuff before this action is ran.

Reply:
I think I'd prefer the non-breaking change, since it allows the flexibility and also gives the right commit message, but in a backwards compatible way

Reply:
Agreed, updated